system/classes/Kohana/Request/Client/Internal.php

```diff
@@ -112,8 +112,8 @@ public function execute_request(Request $request, Response $response)
 		}
 		catch (Exception $e)
 		{
-			// Generate an appropriate Response object
-			$response = Kohana_Exception::_handler($e);
+			// Kaboodle customisation
+			throw $e;
```

Comment:
Yes!

(sorry the latent reply).

At first pass, I just "lift & shift" from the patch file - but I do agree, this can just go.

I'll update the PR